helper/validation/validation_test.go

```diff
@@ -144,6 +144,58 @@ func TestValidationRegexp(t *testing.T) {
 	})
 }
 
+func TestValidateRFC3339TimeString(t *testing.T) {
+	runTestCases(t, []testCase{
+		{
+			val: "2018-03-01T00:00:00Z",
+			f:   ValidateRFC3339TimeString,
+		},
+		{
+			val: "2018-03-01T00:00:00-05:00",
+			f:   ValidateRFC3339TimeString,
+		},
+		{
+			val: "2018-03-01T00:00:00+05:00",
+			f:   ValidateRFC3339TimeString,
+		},
+		{
+			val:         "03/01/2018",
+			f:           ValidateRFC3339TimeString,
+			expectedErr: regexp.MustCompile(regexp.QuoteMeta(`cannot parse "1/2018" as "2006"`)),
```

Comment:
This error message confuses me... what does "Cannot parse ... as 2006" mean? Is this just Go exposing its weird Go formatting representation?

I'd ideally like to try to do better than this and make the messages actually understandable to people who aren't Go experts, but I'm not sure how feasible that is based on what `time.Parse` returns. If this is the quality of the error messages from this function then I'd honestly rather just say "an RFC3339 timestamp is required" for all cases, since that at least offers _some_ reasonable resolution (go and read RFC3339).

Perhaps we could improve on that generic message in _some_ cases by some preprocessing:
* If it looks like the user is using wrong delimiters for either the date or the time, indicate that.
* If the timestamp is at the wrong granularity (only a date where a full time is required), indicate that.

[The `ParseError` type](https://godoc.org/time#ParseError) suggests that we _might_ be able to get some help with recognizing these situations from the `time` package itself, but the fields there are not documented so I'm not really sure what they mean and whether there's enough information here to recognize some common error cases. Maybe you can take a look at what's populated in `ParseError` for a few of these cases and see if it seems like something we could use to produce user-friendly messages for these scenarios?